dts/bindings/clock/st,stm32u3-msi-clock.yaml

```
@@ -0,0 +1,29 @@
+# Copyright (c) 2025 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: STM32U3 Multi Speed Internal Clock
+
+compatible: "st,stm32u3-msi-clock"
+
+include:
+  - name: st,stm32-msi-clock.yaml
+    property-blocklist:
+      - msi-range
+
+properties:
+
+  msi-range:
+    default: 0
+    required: true
+    type: int
+    description: |
+      MSI clock ranges
```

Nit: no need for multi-line description if it's small
```suggestion
    description: MSI clock ranges
```